src/nxt_php_sapi.c

```diff
@@ -411,10 +422,63 @@ nxt_php_setup(nxt_task_t *task, nxt_process_t *process,
         nxt_php_set_options(task, value, ZEND_INI_USER);
     }
 
+    zend_hash_init(&nxt_php_user_config_cache, 0, NULL, (dtor_func_t) nxt_php_user_config_cache_entry_dtor, 1);
+
     return NXT_OK;
 }
 
 
+static void
+nxt_php_activate_user_config(nxt_php_run_ctx_t ctx) {
+    char                     *ptr;
+    user_config_cache_entry  *new_entry, *entry;
+    nxt_time_t               request_time = (nxt_time_t) sapi_get_request_time();
+    char                     path[ctx.script_filename.length + 1];
+
+    nxt_memcpy(path, ctx.script_filename.start, ctx.script_filename.length);
+
+#ifdef NXT_PHP7
+    if ((entry = zend_hash_str_find_ptr(&nxt_php_user_config_cache, path, ctx.script_filename.length)) == NULL) {
+#else /* PHP 5. */
+    if (zend_hash_find(&nxt_php_user_config_cache, path, ctx.script_filename.length + 1, (void **) &entry) == FAILURE) {
+#endif
+        new_entry = nxt_malloc(sizeof(user_config_cache_entry));
+        if (nxt_slow_path(new_entry == NULL)) {
+            return;
+        }
+
+        new_entry->expires = 0;
+        new_entry->user_config = (HashTable *) nxt_malloc(sizeof(HashTable));
+        if (nxt_slow_path(new_entry->user_config == NULL)) {
+            return;
+        }
+
+        zend_hash_init(new_entry->user_config, 0, NULL, (dtor_func_t) config_zval_dtor, 1);
+#ifdef NXT_PHP7
+        entry = zend_hash_str_update_ptr(&nxt_php_user_config_cache, path, ctx.script_filename.length, new_entry);
+#else /* PHP 5. */
+        zend_hash_update(&nxt_php_user_config_cache, path, ctx.script_filename.length + 1, new_entry, sizeof(user_config_cache_entry), (void **) &entry);
+#endif
+    }
+
+    if (request_time > entry->expires) {
+        zend_hash_clean(entry->user_config);
+        ptr = path + (ctx.root->length - 1);
+
+        while ((ptr = strchr(ptr, DEFAULT_SLASH)) != NULL) {
+            *ptr = 0;
+            php_parse_user_ini_file(path, PG(user_ini_filename), entry->user_config);
+            *ptr = '/';
+            ptr++;
+        }
```

Comment:
`DEFAULT_SLASH` macro used to find the slash, and constant `'/'` used to restore the value. It's a bit inconsistent.